test/frontdesk_test.rb

```diff
@@ -0,0 +1,161 @@
+      expect(@front_desk.get_bookings("5 Apr 2020").first).must_be_nil
+    end
+
+    it "returns an empty array if no reservations are found" do
+      expect(@front_desk.get_bookings("5 Apr 2020")).must_equal []
+    end
+  end
+
+  describe "#get_room_bookings" do
+    it "returns an instance of Reservation in a list" do
+      @front_desk.reserve_room(@range)
+      expect(@front_desk.get_room_bookings(1, @range).first).must_be_instance_of Hotel::Reservation
+    end
+  end
+
+  describe "#get_available_rooms" do
+    it "returns an instance of a Room in a list" do
+      @front_desk.reserve_room(@range)
+      expect(@front_desk.get_available_rooms(@range).first).must_be_instance_of Hotel::Room
+    end
+
+    it "returns an array with length equal to the number of available rooms" do
+      18.times do |i|
+        reservation = Hotel::Reservation.new(
+          rooms: Hotel::Room.new(i+1), 
+          date_range: @range)
+        @front_desk.reserve_room(@range)
+      end
+      expect(@front_desk.get_available_rooms(@range).length).must_equal 2
+    end
+
+    it "returns an array with length 20 if no reservations made" do
+      expect(@front_desk.get_available_rooms(@range).length).must_equal 20
+    end
+
+    it "returns the first room if there is no reservation made for it" do
+      expect(@front_desk.get_available_rooms(@range).first.number).must_equal 1
+    end
+  end
+
+  describe "#reserve_room" do
+    it "returns an instance of a new Reservation" do
+      expect(@front_desk.reserve_room(@range)).must_be_instance_of Hotel::Reservation
+    end
+
+    it "returns a new Reservation" do
+      check_in = Date.parse("1 Apr 2020")
+      check_out = Date.parse("3 Apr 2020")
+      expect(@front_desk.reserve_room(@range).rooms.first).must_be_instance_of Hotel::Room
+      expect(@front_desk.reserve_room(@range).date_range.start_date).must_equal check_in
+      expect(@front_desk.reserve_room(@range).date_range.end_date).must_equal check_out
+      expect(@front_desk.reserve_room(@range).cost).must_equal 400
+    end
+
+    it "adds a new Reservation to the list of reservations" do
+      check_in = Date.parse("1 Apr 2020")
+      check_out = Date.parse("3 Apr 2020")
+      @front_desk.reserve_room(@range)
+      expect(@front_desk.reservations.size).must_equal 1
+      expect(@front_desk.reserve_room(@range).rooms.first.number).must_equal 2
+      expect(@front_desk.reserve_room(@range).rooms.first).must_be_instance_of Hotel::Room
+      expect(@front_desk.reserve_room(@range).date_range.start_date).must_equal check_in
+      expect(@front_desk.reserve_room(@range).date_range.end_date).must_equal check_out
+      expect(@front_desk.reserve_room(@range).cost).must_equal 400
+    end
+
+    it "raises and Argument Error if there are no available rooms for provided dates" do
+      20.times do
+        @front_desk.reserve_room(@range)
+      end
+      expect{@front_desk.reserve_room(@range)}.must_raise ArgumentError
```

Comment:
Nice work dealing elegantly with and testing this edge case! Consider creating a custom exception for this type of Error.